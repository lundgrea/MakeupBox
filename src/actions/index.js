export const handleErrors = error => ({
  type: "HANDLE_ERROR",
  error
})

export const isLoading = bool => ({
  type: 'IS_LOADING',
  bool
})

export const getResults = (results) => ({
  type: 'GET_RESULTS',
  results
})

export const setResponses = response => ({
  type: 'SET_RESPONSES',
  response
})