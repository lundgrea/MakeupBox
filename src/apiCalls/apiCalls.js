export const fetchAllMakeup = async () => {
  const url = 'http://makeup-api.herokuapp.com/api/v1/products.json'
  const response = await fetch(url);
  if(!response.ok) {
    throw new Error('There was an error getting makeup')
  }
  const allMakeup = await response.json();
  return allMakeup
}

// export const fetchQuestionOne = async (answer1) => {
//   const url = ``;
//   const response = await fetch(url)
// }

