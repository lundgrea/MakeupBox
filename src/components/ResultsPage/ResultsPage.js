import React from 'react';
import './ResultsPage.css';

export const ResultsPage = () => {
  return (
    <h3>results</h3>
  )
}

export default ResultsPage