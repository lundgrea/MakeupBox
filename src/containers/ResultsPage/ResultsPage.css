
.results-page-headline {
  text-align: center
}

