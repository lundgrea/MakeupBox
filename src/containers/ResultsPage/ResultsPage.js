import React, { Component } from 'react';
import './ResultsPage.css';
import { cleanedSearch, cleanedSearchResults } from '../../dataCleaner/dataCleaner'
import { connect } from 'react-redux';
import { getResults, isLoading, handleErrors } from "../../actions";
import { fetchMakeup } from '../../apiCalls/apiCalls';
import { CardContainer } from '../../components/CardContainer/CardContainer'


export class ResultsPage extends Component {
 
  async componentDidMount () {
    const searches = cleanedSearch(this.props.responses)
    try {
      this.props.isLoading(true)
      const firstResult = await fetchMakeup(searches[0])
      const cleanedFirstData = await cleanedSearchResults(firstResult)
      this.props.getResults(cleanedFirstData)
      const secondResult = await fetchMakeup(searches[1])
      const cleanedSecondData = await cleanedSearchResults(secondResult)
      this.props.getResults(cleanedSecondData)
      const thirdResult = await fetchMakeup(searches[2])
      const cleanedThirdData = await cleanedSearchResults(thirdResult)
      this.props.getResults(cleanedThirdData)
      this.props.isLoading(false);
    } catch ({message}) {
      this.props.handleErrors(message)
      this.props.isLoading(false)
    }
  }

  render() {
    return (
      <section className="results-page">
        <h3 className="results-page-headline">Your Results</h3>
        {this.props.loading && <p className="loading-sentence">Gathering Your Selects</p>}
        {this.props.errors && <p className="loading-sentence">{this.props.errors}</p>}
        {this.props.loading && <div className="loading-div"><img alt='loading' className='loading-animation' src="https://image21.net/103/loading_gif_png/loading_gif_png_59.gif"></img></div>}
        {!this.props.loading && <CardContainer category={this.props.responses} data={this.props.results}/>}
    </section>
    )
  }
}

export const mapStateToProps = store => ({
  errors: store.hasErrored,
  responses: store.responses,
  results: store.results,
  loading: store.isLoading
})

export const mapDispatchToProps = dispatch => ({
  isLoading: bool => dispatch(isLoading(bool)),
  handleErrors: error => dispatch(handleErrors(error)),
  getResults: results => dispatch(getResults(results))
});

export default connect(mapStateToProps, mapDispatchToProps)(ResultsPage)


