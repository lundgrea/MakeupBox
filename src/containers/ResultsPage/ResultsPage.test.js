import React from 'react';
import { shallow } from 'enzyme';
import { ResultsPage, mapStateToProps, mapDispatchToProps } from './ResultsPage';
import { handleErrors, isLoading, getResults } from '../../actions'
import { fetchMakeup } from '../../apiCalls/apiCalls'

jest.mock('../../apiCalls/apiCalls');

describe('ResultsPage', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<ResultsPage />);
    expect(wrapper).toMatchSnapshot();
  }); 

  // it('should call fetchMakeup after mounting', () => {    
  //   const wrapper = shallow(<ResultsPage />)
  //   expect(fetchMakeup).toHaveBeenCalled()
  // })
});


describe('mapStateToProps', () => {
  it('should return the quiz responses list ', () => {
    const mockState = {
      hasErrored: '',
      isLoading: false,
      results: [[{makeup1: '1'}, {makeup2: '2'}], [{makeup3: '3'}, {makeup4: '4'}], [{makeup5: '5'},{makeup6: '6'}] ],
      responses: ['lips', 'bold', 'cruelty-free', 'none']
    }; 
    const expected = {
      errors: '',
      loading: false,
      results: [[{makeup1: '1'}, {makeup2: '2'}], [{makeup3: '3'}, {makeup4: '4'}], [{makeup5: '5'},{makeup6: '6'}] ],
      responses: ['lips', 'bold', 'cruelty-free', 'none']
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

  it('should return the results from the fetch in an array', () => {
    const mockState = {
      hasErrored: '',
      isLoading: false,
      results: [[{makeup1: '1'}, {makeup2: '2'}], [{makeup3: '3'}, {makeup4: '4'}], [{makeup5: '5'},{makeup6: '6'}] ],
      responses: ['lips', 'bold', 'cruelty-free', 'none']
    }; 
    const expected = {
      errors: '',
      loading: false,
      results: [[{makeup1: '1'}, {makeup2: '2'}], [{makeup3: '3'}, {makeup4: '4'}], [{makeup5: '5'},{makeup6: '6'}] ],
      responses: ['lips', 'bold', 'cruelty-free', 'none']
    }; 
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

  it('should return the loading status of the application', () => {
    const mockState = {
      hasErrored: '',
      isLoading: false,
      results: [[{makeup1: '1'}, {makeup2: '2'}], [{makeup3: '3'}, {makeup4: '4'}], [{makeup5: '5'},{makeup6: '6'}] ],
      responses: ['lips', 'bold', 'cruelty-free', 'none']
    }; 
    const expected = {
      errors: '',
      loading: false,
      results: [[{makeup1: '1'}, {makeup2: '2'}], [{makeup3: '3'}, {makeup4: '4'}], [{makeup5: '5'},{makeup6: '6'}] ],
      responses: ['lips', 'bold', 'cruelty-free', 'none']
    }; 
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with an isLoading action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = isLoading(false);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.isLoading(false);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with a handleError action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = handleErrors('error message');
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleErrors('error message');
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with getResults action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = getResults([[{makeup1: '1'}, {makeup2: '2'}], [{makeup3: '3'}, {makeup4: '4'}], [{makeup5: '5'},{makeup6: '6'}] ]);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getResults([[{makeup1: '1'}, {makeup2: '2'}], [{makeup3: '3'}, {makeup4: '4'}], [{makeup5: '5'},{makeup6: '6'}] ]);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});





