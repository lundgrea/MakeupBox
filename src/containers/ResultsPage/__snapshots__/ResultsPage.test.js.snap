// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage should match the snapshot 1`] = `
<section
  className="results-page"
>
  <h3
    className="results-page-headline"
  >
    Your Results
  </h3>
  <p
    className="loading-sentence"
  >
    Gathering Your Selects
  </p>
  <div
    className="loading-div"
  >
    <img
      alt="loading"
      className="loading-animation"
      src="https://image21.net/103/loading_gif_png/loading_gif_png_59.gif"
    />
  </div>
</section>
`;
